Network/HPACK/Table/Dynamic.hs

```diff
@@ -228,6 +228,7 @@ newDynamicTable maxsiz info = do
 
 -- | Renewing 'DynamicTable' with necessary entries copied.
 renewDynamicTable :: Size -> DynamicTable -> IO ()
+renewDynamicTable 0 _ = return ()
```

Comment:
I am still worried about this WA.
Under which circumstances `max table size` should be changed?
```
6.3.  Dynamic Table Size Update

     0   1   2   3   4   5   6   7
   +---+---+---+---+---+---+---+---+
   | 0 | 0 | 1 |   Max size (5+)   |
   +---+---------------------------+

               Figure 12: Maximum Dynamic Table Size Change

   A dynamic table size update starts with the '001' 3-bit pattern,
   followed by the new maximum size, represented as an integer with a
   5-bit prefix (see Section 5.1).
```

How does it related with implementation?
```
isTableSizeUpdate :: Word8 -> Bool
isTableSizeUpdate w = w .&. 0xe0 == 0x20
```

Meantime, tomorrow I will add comment to function as you requested, something like: 

```
-- FIXME: handle case 'Maximum dynamic table size changed to 0'.
```

Reply:
Thanks. I think any http2 endpoint can decide to change dynamic table size at anytime. However, I guess an implementation might decide to do so for either effectively disabling or flushing the table then re-filling it. Whatever happens, in current implementation `http2` fails for both use cases. For disabling it: this WA will solve the problem in case an endpoint effectively disables but not in the other use case. I'd say it's OKay, not great.
I have a number of deadline and then a long vacation break this year, but I plan to spend a chunk of the summer improving my http2/grpc libs. Will spend time on this bug if we have not found better before.

Reply:
Agree, 289d3a0 should resolve this conversation.